TempSensor.ino

```diff
@@ -142,27 +142,102 @@ uint16_t MLX90614::read16(uint8_t a) {
 
 MLX90614 mlx = MLX90614();
 
+const int refreshTempReading = 500;
+const int averagingWindow = 10;
+
 // initialize the library by associating any needed LCD interface pin
 // with the arduino pin number it is connected to
 const int rs = 12, en = 11, d4 = 5, d5 = 4, d6 = 3, d7 = 2;
 LiquidCrystal lcd(rs, en, d4, d5, d6, d7);
 
+class RotatingCounter {
+  private:
+  	int m_counter = 0;
+    bool m_first_pass = true;
+  	int m_max_value = 0;
+  public:
+  	RotatingCounter(int max_value) {
+      m_max_value = max_value;
+    }
+  	void resetCounter(){
+      m_counter = 0;
+      m_first_pass = true;
+    }
+    void increment(){
+      m_counter++;
+      if (m_counter > m_max_value){
+        m_counter = 0;
+        m_first_pass = false;
+      }
+    }
+    int isFirstPass() {
+      return m_first_pass;
+    }
+    int getCounterValue() {
+      return m_counter;
+    }
+};
+
+class MovingAverage {
+  private:
+  	int * m_values;
+    long m_sum = 0;
+    RotatingCounter * m_rotating_counter;
+  public:
+    MovingAverage() {
+      m_values = new int[averagingWindow];
+      m_rotating_counter = new RotatingCounter(10);
```

Comment:
I'd suggest not using the 'new' keyword, and using a static memory allocation in the form of arrays instead.

Reply:
Sure I'll address this one.